### ISSUE TITLE
Autogenerated POJOs under Turkish locale

### ISSUE THREAD

**Issue by tJeyhun** [jOOQ/jOOQ#13100]
Problems with generating POJOs and mapping 

Problem : locale setting in pom.xml

Given table 

    CREATE TABLE aktarmalar (
        veritabani_id uuid NOT NULL,
        kullanici_id uuid NOT NULL,
        abonelik_id uuid NOT NULL,
        bolge_id uuid NOT NULL,
        baslama_tarih_saat timestamp NOT NULL,
        CONSTRAINT pk_aktarmalar PRIMARY KEY (veritabani_id)
    );

locale set in pom.xml

    <target>
      <packageName>com.ekser.onlinehukuk.drs.jooq</packageName>
      <directory>jooq</directory>
      <locale>en</locale>
    </target>

`mvn jooq-codegen:generate` generates pojo like

    public class Aktarmalar implements Serializable {
    ...
        private UUID          veritabaniİd; // turkish İ character 
        private UUID          kullaniciİd; // turkish İ character
        private UUID          abonelikİd; // turkish İ character
        private UUID          bolgeİd; // turkish İ character
        private LocalDateTime baslamaTarihSaat;
    ...

but it should

    public class Aktarmalar implements Serializable {
    ...
        private UUID          veritabaniId; // latin uppercase for 'i' character
        private UUID          kullaniciId; // latin uppercase for 'i' character
        private UUID          abonelikId; // latin uppercase for 'i' character
        private UUID          bolgeId;  // latin uppercase for 'i' character
        private LocalDateTime baslamaTarihSaat;
    ...

setting locale from pom.xml not applied.

under english locale, i.e. "-Duser.language=en" for maven and application it works just fine.

Same problem continues under application, for example `fetchInto(Aktarmalar)` does not map bogus fields, event when new `jooq.conf.Setting` given: `new Settings().withLocale(Locale.ENGLISH)`

I am confused because I thought all these locale settings were intended to fix issues with generating and mapping under different locales. Especially Turkish locale is kind of unique.

http://haacked.com/archive/2012/07/05/turkish-i-problem-and-why-you-should-care.aspx/

### Versions

- jOOQ: 3.16.4
- Java: Jdk11
- Database (include vendor): Postgres 12
- OS: Windows 10 Pro Turkish

**Comment by lukaseder** [comment]
Thanks for your message. Can you please provide a full reproducer that helps reproduce this on my side? I'm suspecting that your calling

> `mvn jooq-codegen:generate` generates pojo like

Might cause the problem. There's a subtle difference between putting the `<configuration>` in the `<plugin>` or in the `<execution>` of the plugin.

Here's a useful template to help create reproducers: https://github.com/jOOQ/jOOQ-mcve

**Comment by tJeyhun** [comment]
I have tried changing pom as you suggested, same result. Maybe I am missing something. 

SQL table 

    CREATE TABLE public.aktarmalar (
	    veritabani_id uuid NOT NULL,
	    kullanici_id uuid NOT NULL,
	    abonelik_id uuid NOT NULL,
	    bolge_id uuid NOT NULL,
	    baslama_tarih_saat timestamp NOT NULL,
	    dinamik_surum vch020 NOT NULL COLLATE "tr-TR-x-icu",
	    bitis_tarihi timestamp NULL,
	    islem_baslatildi bool NULL,
	    islem_bitti bool NULL,
	    islem_heart_beat timestamp NULL,
	    CONSTRAINT pk_aktarmalar PRIMARY KEY (veritabani_id)
    );

pom.xml attached

[pom.zip](https://github.com/jOOQ/jOOQ/files/8095796/pom.zip)

for reproducing you may want to set maven `user.language` under `.mvn/jvm.config` file
like `-Duser.language=tr`

**Comment by tJeyhun** [comment]
Just wanted to say that this problem is really really annoying. I love JOOQ and using it in 3 commercial products. It is not a problem to change locale when running backend server applications. But I also have a desktop application built on Eclipse RCP with JavaFX and support for Turkish locale is essential.

Thanks

**Comment by lukaseder** [comment]
Thanks for your patience. I understand that this is an annoying problem. I imagine that it is extremely frequent when developing things in Turkish. It's obviously too late to recommend using English table and column names instead, though as a German speaker, I've always worked around similar problems (mostly encoding, not locale specific) by simply not using any German in any software, ever.

Anyway, as I said, the above is obviously not an option, so let's look at what needs to be fixed. Can you please also provide an example of the wrong runtime behaviour? What are you doing *specifically* to produce this? (I understand that setting up an MCVE is a bit more time consuming, but to be sure I fix the *exact* issue you're experiencing, I will need to see the *exact* issue, so I have to reproduce it)

**Comment by tJeyhun** [comment]
Thanks for the fast replying. 

You can see in the table DDL example that I have supplied, that no Turkish characters were used. The Turkish alphabet in general is the same Latin. Except for one case: lowercase `i` is uppercased as `İ` (and vice versa), and uppercase `I` is lowercased as `ı` (and vice versa). So my tables are just normal as it is.

Because generating POJOs and Records with `İ` characters in field names is not acceptable, as for field named `field_id` it generates `getFieldİd` and `setFieldİd` as setter and getter. So I'm using English locale for maven. It gives me the right result.  Now field with name `field_id` is generated as `getFieldId` and `setFieldId` as expected. The problem arise under application when I use something like

    dsl.selectFrom(AKTARMALAR) // table
        .where(AKTARMALAR.ABONELIK_ID.eq(myId))
        .and(AKTARMALAR.BASLAMA_TARIH_SAAT.notNull)
        .and(AKTARMALAR.BITIS_TARIHI.notNull)
        .and(AKTARMALAR.ISLEM_BASLATILDI.^false)
        .fetchInto(Aktarmalar) // generated POJO

As this code uses reflection and silently skips missing fields, under Turkish locale it still gives me POJO, but fields that contain `i` character with preceding `_` underscore (veritabani_id, abonelik_id, etc.) are null. So I have to change my application locale with `-Duser.language=en` to English. Then all problems disappear. POJO is populated with values.

In short, I have to generate and use POJOs under the English locale for correct behavior.

Thanks

**Comment by lukaseder** [comment]
> You can see in the table DDL example that I have supplied, that no Turkish characters were used. The Turkish alphabet in general is the same Latin. Except for one case: lowercase `i` is uppercased as `İ` (and vice versa), and uppercase `I` is lowercased as `ı` (and vice versa). So my tables are just normal as it is.

I see, thanks for the clarification.

I will try to reproduce the runtime problem.
